File: TLE/TLE '16 Contest 8 P1 - Paper Hole Punching.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    
  string t;
  cin>>t;
  int n;
  cin>>n;
    
  for(int i = 0; i < n; i++){
    string x;
    cin>>x;
    bool a = true;
    for(int j = 0; j < t.length(); j++){
      if(x.find(t[j]) > x.length()){
        cout<<"no"<<endl;
        a = false;
        break;
      }
    }

    if(a){
      cout<<"yes"<<endl;
    }
        
  }
    
  return 0;
}

File: TLE/TLE '16 Contest 8 P1 - Paper Hole Punching.py
t = str(input())
n = int(input())

for i in range(n):
  x = str(input())
  a = True
  for j in range(len(t)):
    if t[j] not in x:
      a = False
      print("no")
      break
  if a:
    print("yes")